.github/workflows/ci-api-proto.yml

```
@@ -57,6 +57,15 @@ jobs:
               event: 'REQUEST_CHANGES',
               body: 'You have altered the generated proto files but they do not match what is expected.\nPlease run "script/api_protobuf/api_protobuf.py" and commit the changes.'
             })
+      - if: failure()
+        name: Show changes
+        run: git diff
+      - if: failure()
+        name: Archive artifacts
+        uses: actions/upload-artifact@v4
```

Just my personal preference to use the full version
```suggestion
        uses: actions/upload-artifact@v4.6.2
```